tools/base/entry_point.py

```diff
@@ -1,10 +1,26 @@
+import pathlib
 import subprocess
 import sys
 
+ENTRY_POINT_ALIAS = "_ENTRY_POINT_ALIAS_"
+
+
+def find_tool_path():
+    entry_point_alias = ENTRY_POINT_ALIAS.split("/external/")[1]
+
+    for x in pathlib.Path(".").glob(f"**/{entry_point_alias}"):
```

Comment:
This seems a bit of a wide net to cast. PR LGTM otherwise.

Reply:
the entry point alias that its matching is a pretty long string - if it matches we can be fairly sure we found the binary

i couldnt find any other way to do it - not sure why but the location im getting back from the alias wasnt matching exactly,  even tho the tool is actually there (on a slightly different filepath)

Reply:
Can you provide some concrete path examples? I think it's probably OK as a workaround, but it just has a code smell to it, where we might end up with some hard to diagnose bug later due to this use of globbing.

Reply:
>  it just has a code smell to it

yep, agreed, i spent quite a few hours on this, and didnt find an alternative - so reluctantly did it this way - ill dump what is being mangled now and add as a comment...

Reply:
hmm, not sure about adding a comment as the lines are very long, ill post here for now and we can figure out

essentially there are 2 situations that it needs to work in - when its called directly with `bazel run //path/to:entry_point` and where are a build rule calls it - ie `bazel build //rule/that/calls:entry_point`

This is the before/after mangling for the first - ie bazel run

```console
ENTRY_POINT_ALIAS: bazel-out/k8-opt-exec-2B5CBBC6/bin/external/base_pip3_envoy_docs_sphinx_runner/rules_python_wheel_entry_point_envoy.docs.sphinx_runner
FOUND PATH: external/base_pip3_envoy_docs_sphinx_runner/rules_python_wheel_entry_point_envoy.docs.sphinx_runner

```

And this is for the second:

```console

ENTRY_POINT_ALIAS: bazel-out/host/bin/external/base_pip3_envoy_docs_sphinx_runner/rules_python_wheel_entry_point_envoy.docs.sphinx_runner
FOUND PATH: bazel-out/host/bin/tools/docs/sphinx_runner.runfiles/base_pip3_envoy_docs_sphinx_runner/rules_python_wheel_entry_point_envoy.docs.sphinx_runner

```

Reply:
ive cleaned the macro up a little, but whatever way i look at it the path that the genrule sees is different from the path that the py_binary sees 

im pretty confident that this workaround works, not sure how else to achieve the desired outcome

Reply:
OK. The only other suggestions I would have here are to think of (1) passing the path in via command-line from Bazel which should know what it is or (2) using Bazel query. @lizan might also have some insight. In any case, it's probably not worth spending a ton of time optimizing, so maybe just document why this is the way it is and we can land.

Reply:
passing via the path wont work - its the reason for shifting to using a template - the arg gets lost when the rule is called from another rule, ill add a comment now...